### ISSUE TITLE
[Feature]: User-provided scripts to show custom formats in SDV

### ISSUE THREAD

**Issue by etienneschalk** [etienneschalk/scientific-data-viewer#91]
### Feature Category

Data Visualization

### Priority

Low - Nice to have

### Feature Description

User-provided scripts to show custom formats in SDV

### Problem Statement

Some formats or products are more specific (eg formats carried by other formats, like the Sentinel-2 SAFE product format, that contains grib and TIF)

For now one has to manually navigate into the product and open files one by one, whether it's in code or in a file explorer.

### Proposed Solution

The user-provided script could require the following condition to be met: it should provide a xarray-backend. This backend would only be used by `get_data_info` to load the data into a DataTree.

The user could register specific patterns of filenames and extension to trigger their custom scripts

An option should also be left to try to open any file (eg right click on the file > context menu > try to open with SDV > [collection of scripts])

### Use Cases

Working with satellite data 

### Code of Conduct

- [x] I have searched existing issues and this is not a duplicate
- [x] I have read and agree to follow the [Contributing Guidelines](CONTRIBUTING.md)

**Comment by etienneschalk** [comment]
Studying the closing as Not Planned because: 

(1) The extension uses xarray. xarray can be be extended by backends. 

⚠️  Currently the list of supported backends is **hardcoded**. The choice was made not to use xarray auto-backend detection for better managing the various frequent formats. However, a less-known backend might not be usable by the extension even if it was available! The extension should use xarray backend availability check strategy, and try at least the best effort to use less frequent backends. Optionnally, we could also add a user-configurable list of formats to backends. 

(2) About private backends : a user can install a private backend in their own python environment. However, with uv, this would not be possible easily, since the uv env uses only well known formats. 

In all cases, the better backend detection should be integrated into a python refactoring task. Indeed the current scripts became too big with latest versions, and the need for a "scientific-data-viewer" package, associated with this project, published on PyPI (or at least contained and install locally in the extension's own environnement) rises. The only issue is that, initially, I did not wanted to install anything else than well-known libraries in the user env. Using scripts was more liteweight.. So it would be nice for uv env, but not nice for using the python extension env
